Frontend/src/CustomScripts/convertData.mjs

```diff
@@ -58,37 +74,53 @@ for (const key of Object.keys(SC1DataFormat)) {
     max,
   };
 
-  if (categoryIdx === -1) {
-    // doesn't exist = create new category
-    categories.push({
-      category,
-      values: [listing],
-    });
+  let res = category.split(';')
+
+  let c = res[0]
+  let subc = res[1]
+  let ind = undefined;
+  for (let j = 0; j < newCategories.length; j++) {
+    if (newCategories[j].category.localeCompare(c) == 0) {
+      ind = j
+      break
+    }
+  }
+
+  if (ind) {
+    let todo = true
+    for (let i = 0; i < newCategories[ind].subcategories.length; i++) {
+      if (newCategories[ind].subcategories[i].subcategory.localeCompare(subc) == 0) {
+        todo = false
+        newCategories[ind].subcategories[i].values.push(listing)
+      }
+    }
+    if (todo){
+      newCategories[ind].subcategories.push({
+        'subcategory': subc,
+        'values': [listing]
+      })
+    }
```

Comment:
```suggestion
    const subInd = newCategories[ind].subcategories.findIndex((sc) => sc.subcategory === subc);
    
    if (subInd !== -1) {
      newCategories[ind].subcategories[subInd].values.push(listing)
    } else {
      newCategories[ind].subcategories.push({
        'subcategory': subc,
        'values': [listing]
      })
    }
```
Can make this more compact by using `findIndex()` like mentioned above

Reply:
fixed